video/out/vo_gpu_next.c

```diff
@@ -874,13 +874,33 @@ static void draw_frame(struct vo *vo, struct vo_frame *frame)
     bool cache_frame = will_redraw || frame->still;
     bool can_interpolate = opts->interpolation && frame->display_synced &&
                            !frame->still && frame->num_frames > 1;
+    double pts_offset = can_interpolate ? frame->ideal_frame_vsync : 0;
     params.info_callback = info_callback;
     params.info_priv = vo;
     params.skip_caching_single_frame = !cache_frame;
     params.preserve_mixing_cache = p->inter_preserve && !frame->still;
     if (frame->still)
         params.frame_mixer = NULL;
 
+    // pl_queue advances its internal virtual PTS and cull the available frames
+    // based on this value and VPS/FPS ratio. Requesting non-monotonic PTS is
+    // invalid usage, reset if this happens in an atempt to recover as much as
+    // possible. This should never happen in practice and if it does, should be
+    // fixed. Interpolated may be before current PTS, but in this case we should
+    // have this frame available in pl_queue or want_reset is already requested.
+    // Clamp the check to 0, as we don't have the previous frame anyway in vo_frame.
+    struct pl_source_frame vpts;
+    if (frame->current && !p->want_reset) {
+        if (pl_queue_peek(p->queue, 0, &vpts) &&
+            frame->current->pts + MPMAX(0, pts_offset) < vpts.pts)
+        {
+            MP_VERBOSE(vo, "Forcing queue refill, PTS(%f + %f | %f) < VPTS(%f)\n",
```

Comment:
If this shouldn't ever happen, should this verbosity be raised so people actually see it?

Reply:
I don't want to scary user about it, it is quite spammy log when it happens. User wouldn't be able to fix it anyway and we silently recover from it. Not trying to hide it, but not sure if this should spam user, especially playback will still work.

That said, I initially had `MP_ERR` here and `assert` in the other (clamp) place. But I had change of heart to not be so strict.

Let me know what you think?